hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/AbstractContainerReportHandler.java

```diff
@@ -361,6 +361,7 @@ private void updateContainerReplica(final DatanodeDetails datanodeDetails,
         .setKeyCount(replicaProto.getKeyCount())
         .setReplicaIndex(replicaProto.getReplicaIndex())
         .setBytesUsed(replicaProto.getUsed())
+        .setEmpty(replicaProto.getIsEmpty())
```

Comment:
Thanks for the review @sodonnel, I have handled this scenario now.

Reply:
I'm not sure we should have removed this change, as I think we should try to be compatible between mixed versions, as we don't know when rolling upgrades may happen, or what someone will try for upgrading.

Reply:
Agree, If rolling upgrade is planned now or in future this can impact. I think we can handle this change now. This change is an extra precaution and doesn't impact anything else.